### PR TITLE
Refactor basket tasks to take `newsletters` as arg. 

### PR DESCRIPTION
**Pull request by johngian**
* Extend tasks to take `newsletters` as arg
* Add new `settings` entry for NDA basket
* Add extra admin actions: subscribe/unsubscribe nda/vouched basket
* When user changes email re-subscribe to all newsletters not only vouched

### CODE REVIEW

**Comment by johngian** [comment]
@akatsoulas or @comzeradd r?

File: mozillians/settings/base.py [hunk 1]
```diff
@@ -437,7 +437,9 @@ def COMPRESS_JINJA2_GET_ENVIRONMENT():
 else:
     # Basket requires SSL now for some calls
     BASKET_URL = 'https://basket.mozilla.com'
-BASKET_NEWSLETTER = 'mozilla-phone'
+
+BASKET_NEWSLETTER_VOUCHED = 'mozilla-phone'
```
Comment (akatsoulas) [inline mozillians/settings/base.py]: I would go with something like `VOUCHED_NEWSLETTER` and `NDA_NEWSLETTER`

Comment (johngian) [inline mozillians/settings/base.py]: I prefer to have them prefixed. We can always swap the last 2 words.

Comment (akatsoulas) [inline mozillians/settings/base.py]: :+1:

File: mozillians/groups/models.py [hunk 2]
```diff
@@ -278,7 +279,8 @@ def add_member(self, userprofile, status=GroupMembership.MEMBER):
                     if membership.status in [GroupMembership.PENDING, GroupMembership.MEMBER]:
                         if self.functional_area:
                             # Group is functional area, we want to sent this update to Basket.
-                            update_basket_task.delay(userprofile.id)
+                            newsletters = [settings.BASKET_NEWSLETTER_VOUCHED]
```
Comment (akatsoulas) [inline mozillians/groups/models.py]: You can have this var outside the `if` statement and use it in both `if` and `else` statement

File: mozillians/phonebook/management/commands/check_basket.py [hunk 3]
```diff
@@ -21,7 +21,8 @@ def handle(self, *args, **options):
         # The tasks module looks these up at import time, and might get BASKET_API_KEY
         # from the environment rather than settings if it's there, so look directly at
         # what values the tasks module ended up with.
-        required_settings = ['BASKET_API_KEY', 'BASKET_NEWSLETTER', 'BASKET_URL']
+        required_settings = ['BASKET_API_KEY',
+                             'BASKET_URL']
```
Comment (akatsoulas) [inline mozillians/phonebook/management/commands/check_basket.py]: No need to break this line

File: mozillians/phonebook/tests/test_check_basket.py [hunk 4]
```diff
@@ -65,11 +64,6 @@ def test_no_url(self):
         with patch('mozillians.users.tasks.BASKET_URL', new=None):
             ok_(not self.run_check())
 
-    def test_no_newsletter(self):
```
Comment (akatsoulas) [inline mozillians/phonebook/tests/test_check_basket.py]: Why did you remove this check?

**Comment by johngian** [comment]
Closing this one after #1467
